### PR TITLE
Fixes ChatScreen calling .setScreen

### PR DESCRIPTION
**Pull request by RealMangorage**
Fixes ChatScreen calling .setScreen when we already did .setScreen via ClientCommand for example.

Old Behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java then proceeds to .setScreen to null

New behavior:
/cmd openScreen -> 
Opens screen
ChatScreen.java will only .setScreen to null if minecraft.screen == this (ChatScreen)

### CODE REVIEW

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Tested to work with a small modification to `ClientCommandTest` that adds a subcommand that runs `Minecraft.getInstance().setScreen(new PauseScreen(false))`.

However, it seems that `InBedChatScreen` has the same unpatched call in its `keyPressed` method. That should also be patched with this fix, unless someone can provide a reason to leave it unpatched.

**Comment by RealMangorage** [comment]
I moved the check into .handleChatInput() for minimal code changes.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Based on the patch hunk header and the CI patch application failure, that patch ain't right.

Please regenerate the patch, and commit it properly rather than using the GitHub web editor.

**Comment by RealMangorage** [comment]
Okay

File: patches/minecraft/net/minecraft/client/gui/screens/ChatScreen.java.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/net/minecraft/client/gui/screens/ChatScreen.java
++++ b/net/minecraft/client/gui/screens/ChatScreen.java
+@@ -225,7 +_,7 @@
+             this.f_96541_.f_91074_.f_108617_.m_246175_(p_242400_);
+          }
+ 
+-         return true;
++         return f_96541_.f_91080_ == this; // To Prevent closing the Screen if it isn't Chat Screen
```
Comment (sciwhiz12) [inline patches/minecraft/net/minecraft/client/gui/screens/ChatScreen.java.patch]: This marker comment should be prefixed with `FORGE:`, and the capitalization be normalized to sentence case.
```suggestion
+         return f_96541_.f_91080_ == this; // FORGE: Prevent closing the screen if another screen has been opened
```

File: src/main/java/net/minecraftforge/server/LanguageHook.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import com.google.gson.Gson;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import com.sun.jna.platform.unix.X11;
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/server/LanguageHook.java]: Please revert this change.

File: src/test/java/net/minecraftforge/PatchScreen.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package net.minecraftforge;
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/PatchScreen.java]: Missing license header, please run the `updateLicenses`/`licenseFormat` task.

Additionally, move this to under the `.debug.client` subpackage please.

File: src/test/java/net/minecraftforge/PatchScreen.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+
+public class PatchScreen extends Screen {
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/PatchScreen.java]: Formatting: braces on newlines, all throughout this class.

File: src/test/java/net/minecraftforge/PatchScreen.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    public PatchScreen(Component p_96550_) {
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/PatchScreen.java]: Please rename all the SRG parameters in this class to proper parameter names.

File: src/test/java/net/minecraftforge/debug/chat/ClientCommandTest.java [hunk 6]
```diff
@@ -104,6 +106,11 @@ private void init(RegisterClientCommandsEvent event)
                                 .executes((context) -> {
                                     context.getSource().sendSuccess(Component.literal("Executed command"), false);
                                     return 1;
+                                }))
+                        .then(Commands.literal("screenpatch")
+                                .executes((stack) -> {
+                                    Minecraft.getInstance().setScreen(new PatchScreen(Component.literal("patch")));
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/chat/ClientCommandTest.java]: To avoid classloading problems with regards to the (implicit) cast of `PatchScreen` to `Screen`, please move this call to `Minecraft#setScreen` to a static method within `PatchScreen`, which this lambda calls.

File: patches/minecraft/net/minecraft/network/CompressionEncoder.java.patch [hunk 7]
```diff
@@ -13,7 +13,7 @@
           friendlybytebuf.m_130130_(0);
           friendlybytebuf.writeBytes(p_129453_);
        } else {
-+         if (!DISABLE_PACKET_DEBUG && i > net.minecraft.network.CompressionDecoder.MAXIMUM_UNCOMPRESSED_LENGTH) {
++         if (!DISABLE_PACKET_DEBUG && i > net.minecraft.network.CompressionDecoder.f_182672_) {
```
Comment (sciwhiz12) [inline patches/minecraft/net/minecraft/network/CompressionEncoder.java.patch:16]: I've no idea how this got past us. @Matyrobbrt, ideas 👀?

File: src/test/java/net/minecraftforge/debug/world/LoginPacketSplitTest.java [hunk 8]
```diff
@@ -66,6 +66,7 @@
  * To test if the packet is too large simply remove the login packet from the {@link net.minecraftforge.network.filters.ForgeConnectionNetworkFilter}
  * and try connecting again. You should see the connection fail.
  */
+/**
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/world/LoginPacketSplitTest.java]: @Matyrobbrt why u cause me pain.

Please fix the compile errors and reinstate this test mod by:
- Changing the call to `FriendlyByteBuf#writeWithCodec` into `FriendlyByteBuf#writeJsonWithCodec`; and
- Changing the reference to `com.mojang.bridge.game.PackType.DATA` to simply `PackType.SERVER_DATA`, importing the new class.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Comment by RealMangorage** [comment]
I pushed a commit that fixes that.

File: src/test/java/net/minecraftforge/debug/client/TestScreen.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.debug.client;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+public class TestScreen extends Screen
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/client/TestScreen.java]: Formatting: space between last import statement and class declaration.
```suggestion
import net.minecraft.network.chat.Component;

public class TestScreen extends Screen
```

File: src/test/java/net/minecraftforge/debug/client/TestScreen.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public static int open() {
+        Minecraft.getInstance().setScreen(new TestScreen(Component.translatable("screen.patch")));
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/client/TestScreen.java]: Formatting: braces on newlines. 

Additionally, use a literal component here with readable text, since there is no definition for the `screen.patch` translation key.

```suggestion
    public static int open()
    {
        Minecraft.getInstance().setScreen(new TestScreen(Component.literal("Test screen")));
```

File: src/test/java/net/minecraftforge/debug/client/TestScreen.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+        return 1;
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/client/TestScreen.java]: Use `Commands.SINGLE_SUCCESS` here.

File: src/test/java/net/minecraftforge/debug/client/TestScreen.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public TestScreen(Component pComponent) {
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/client/TestScreen.java]: Formatting: braces on newlines. Additionally, there shouldn't be a prefix for the parameters.
```suggestion
    public TestScreen(Component title)
    {
```

File: src/test/java/net/minecraftforge/debug/client/TestScreen.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+        super(pComponent);
+    }
+
+    @Override
+    public void render(PoseStack pPoseStack, int pMouseX, int pMouseY, float pPartialTick) {
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/client/TestScreen.java]: Formatting: braces on newlines. Additionally, there shouldn't be a prefix for the parameters.
```suggestion
    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick)
    {
```

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Review by sciwhiz12 — APPROVED** [review]
